Ci script improvements

psychon opened the pull request:
Some stuff, the main thing being that I can run `tests/run.sh` now, too, even though I don't have an up-to-date version of awesome installed.

blueyed commented on tests/run.sh:
```diff
@@ -62,47 +60,68 @@ XDG_CONFIG_HOME="./"
+        if [ $ret = 0 ]; then
+            break
+        fi
+        max_wait=$(expr $max_wait - 1)
```
That needs an `|| true`: `expr 1 - 1` will exit with code 1.
See 938987a1.

blueyed commented on tests/run.sh:
```diff
@@ -62,47 +60,68 @@ XDG_CONFIG_HOME="./"
+            break
+        fi
+        max_wait=$(expr $max_wait - 1)
+        if [ "$max_wait" -lt 0 ]; then
```
..or use `-le 1` here?!

blueyed commented on tests/run.sh:
```diff
@@ -62,47 +60,68 @@ XDG_CONFIG_HOME="./"
-AWESOME_CLIENT="$root_dir/utils/awesome-client"
+# Not in Travis?
+if [ "$CI" != true ]; then
+    # Prepare a config files pointing to a working theme
```
s/files/file/

psychon commented:
I rebased this onto your "add lgi 0.7.0"-branch. My plan is to wait until that other branch is merged and then rebase this onto master again and only merge this then.

Note that I removed your `kill -0` trick again. The timeout handles this situation as well and it does not work for my helper function when it is used via `xrdb` to wait for the X11 server to start up.

psychon commented:
Rebased, can this be merged?
